### PR TITLE
Update stages dockerfile for icubhead-withuser and robometry children

### PR DESCRIPTION
**Pull request by MSECode**
This PR has the following changes:

We realized that the child image of superbuild-icubhead-withuser, i.e. superbuild-icubhead-robometry,  was not using the icubuser but the root user instead.
Therefore, I made some changes on both the parent and the child image thus to update the stages of the compilation and making possible to have the icub user on the child image as well. 
In this way it is possible to export file from the container locally, which is the main aim of the `withuser` images. This can be achieved only using the container with the ownership of a local (not root) user.

cc: @valegagge

### CODE REVIEW

**Comment by MSECode** [comment]
@valegagge 

Check all the modifications.
I've test them locally. If u have any doubt just let me know

File: dockerfile_images/basic/superbuild-icubhead-robometry/Dockerfile [hunk 1]
```diff
@@ -19,9 +27,18 @@ ARG INSTALL_DIR
...
+USER $USERNAME
+
+COPY entrypoint.sh /usr/local/bin/entrypoint.sh
+COPY setup.sh ${ROBOTOLOGY_INITIALIZATION_FILE}
+RUN sudo chown -R icub: /usr/local/bin/entrypoint.sh
+RUN sudo chown -R icub: /usr/local/bin/setup_robotology_tdd.sh
+
 # Some other definitions
 ARG CMAKE_GENERATOR="Unix Makefiles"
 ARG BUILD_TYPE=Release
```
Comment (valegagge) [inline dockerfile_images/basic/superbuild-icubhead-robometry/Dockerfile:44]: Hi @MSECode ,
this is only a question: why did you move the `ARG sbtag`and ``ARG release` on top, while you left BUILD_TYPE here?

**Comment by valegagge** [comment]
Hi @MSECode ,
I have some questions/doubts related to some points, but since this image works on your laptop, I'll merge it in order to avoid blocking the setup preparation. 
In the next few days we'are going to discuss them. 

Thanks for your fix! 🏅

**Comment by MSECode** [comment]
I'll create a meeting then so it's scheduled
